test/libsolidity/SemVerMatcher.cpp

```diff
@@ -159,7 +163,6 @@ BOOST_AUTO_TEST_CASE(negative_range)
 	vector<pair<string, string>> tests = {
 		{"1.0.0 - 2.0.0", "2.2.3"},
 		{"^1.2.3", "1.2.3-pre"},
-		{"^1.2", "1.2.0-pre"},
```

Comment:
The npm semver package still reports this as a non-match. I think `^1.2` should be equivalent to `^1.2.0` and thus not match the prerelease version.

Reply:
Ok, I'll see what I can do.